### PR TITLE
Randao random extension

### PR DESCRIPTION
**Pull request by damianmarti**
Extension implementing a random generator using RANDAO.

I tried to build this as simple and generic as possible, to keep in mind this can be used for different applications that need some randomness.

It's working on Foundry now (Foundry does not set the withdrawalsRoot header) but the random number generated is always the same because the mixHash field from the header (where the randomness is taken from) is always zero.

To install the extension on Foundry you have to use the new create-eth version, which has not been released yet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,21 @@
...
-- [erc-20](https://github.com/scaffold-eth/create-eth-extensions/tree/erc-20): This extension introduces an ERC-20 token contract and demonstrates how to interact with it, including getting a holder balance and transferring tokens.
-
-## Create your own extension
+## 🚀 Setup extension
 
-You can extend Scaffold-ETH by creating your own extension. To do so, you need to create a new repository with the following structure:
+Deploy your contract running `yarn deploy`
 
-`ToDo`
+## Interact with the token
```
Comment (Pabl0cks) [inline README.md:19]: Maybe token -> demo / on-chain randomness

File: README.md [hunk 2]
```diff
@@ -1,29 +1,21 @@
...
 
-This repository holds all the BG curated extensions for [create-eth](https://github.com/scaffold-eth/create-eth), so you can extend the functionality of your Scaffold-ETH project.
+This extension shows how to use on-chain randomness using RANDAO for truly on-chain unpredictable random sources.
 
-## Usage
+Ethereum PoS introduces randomness using block.mixHash (prevRandao). Look at [EIP-4399](https://eips.ethereum.org/EIPS/eip-4399) for more information.
+
+## Installation
 
 You can install any of the extensions in this repository by running the following command:
```
Comment (Pabl0cks) [inline README.md:9]: I think we can delete this sentence and just leave the command

**Review by Pabl0cks — COMMENTED** [review]
Really cool extension @damianmarti !! Only added a couple of nitpicks on the Readme 🙏

**Comment by damianmarti** [comment]
> Really cool extension @damianmarti !! Only added a couple of nitpicks on the Readme 🙏

Thanks!!

File: README.md (hunk 2 above)
Comment (damianmarti) [reply README.md:9]: Yeah, seems irrelevant, I keep it just because we leave this sentence on the README from other extensions.

**Review by technophile-04 — APPROVED** [review]
Thanks @damianmarti and Pablo for testing! Just pushed UI tweaks and minor issues of hydration and unhandled error at https://github.com/scaffold-eth/create-eth-extensions/pull/28/commits/3a80ef8fda907e5714e4ff5aab1c5b1dd28fcf0e and https://github.com/scaffold-eth/create-eth-extensions/pull/28/commits/a9f20afb407cfb257cb3ff92a88a788cf25f3574

But looks great!! Merging this! 

We should update this repo README with exteniosn, add to create-eth core extensions and update scaffold docs

**Pull request merged**